File: qqbot-admin-api/src/main/java/com/mieai/qqbot/admin/plugins/PluginUploadResponse.java
package com.mieai.qqbot.admin.plugins;

import com.mieai.qqbot.plugin.host.PluginArtifactInstallResult;

public record PluginUploadResponse(
        String operation,
        PluginArtifactResponse artifact,
        String previousVersion,
        String previousSha256) {

    static PluginUploadResponse from(PluginArtifactInstallResult result, PluginArtifactResponse artifact) {
        return new PluginUploadResponse(result.operation(), artifact,
                result.previousVersion().orElse(null), result.previousSha256().orElse(null));
    }
}

File: qqbot-module-plugin-support/src/main/kotlin/com/mieai/qqbot/admin/plugins/PluginUploadResponse.kt
package com.mieai.qqbot.admin.plugins

import com.mieai.qqbot.plugin.host.PluginArtifactInstallResult

data class PluginUploadResponse(
    val operation: String,
    val artifact: PluginArtifactResponse,
    val previousVersion: String?,
    val previousSha256: String?,
) {
    companion object {
        fun from(
            result: PluginArtifactInstallResult,
            artifact: PluginArtifactResponse,
        ): PluginUploadResponse = PluginUploadResponse(
            result.operation,
            artifact,
            result.previousVersion,
            result.previousSha256,
        )
    }
}
